### DISCUSSION TITLE
How to select GPU to execute a model in NNAPI EP

### DISCUSSION THREAD

**Discussion by Uwenisme** [microsoft/onnxruntime#17159] [Other Q&A]
I follow the example code to run a model by using NNAPI EP. But I can't find any example to run a model in gpu or npu. How can I select different devices (cpu, gpu or npu) ? Please give me some tips! Thanks a lot!

  Ort::Env onnx_env = Ort::Env{ORT_LOGGING_LEVEL_ERROR, "Default"};
  Ort::SessionOptions so;
  so.SetGraphOptimizationLevel(GraphOptimizationLevel::ORT_ENABLE_ALL);
  uint32_t nnapi_flags = 0;
  nnapi_flags |= NNAPI_FLAG_CPU_DISABLED;
  //nnapi_flags |= NNAPI_FLAG_USE_NCHW;
  Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_Nnapi(so, nnapi_flags));
  Ort::Session session(onnx_env, model_content.data(), fileSize,so);

**Comment by skottmckay** [comment]
We don't have much control over how NNAPI executes the model.

Please see https://github.com/microsoft/onnxruntime/discussions/10692#discussioncomment-6811281
